authenticatorbridge/src/main/java/com/bitwarden/authenticatorbridge/manager/AuthenticatorBridgeManagerImpl.kt

```diff
@@ -0,0 +1,193 @@
+     */
+    private val bridgeServiceConnection = object : ServiceConnection {
+        override fun onServiceConnected(name: ComponentName, service: IBinder) {
+            onServiceConnected(service)
+        }
+
+        override fun onServiceDisconnected(name: ComponentName) {
+            onServiceDisconnected()
+        }
+    }
+
+    override val accountSyncStateFlow: StateFlow<AccountSyncState> =
+        mutableSharedAccountsStateFlow.asStateFlow()
+
+    init {
+        // Listen for lifecycle events
+        processLifecycleOwner.lifecycle.addObserver(
+            object : DefaultLifecycleObserver {
+                override fun onStart(owner: LifecycleOwner) {
+                    bindService()
+                }
+
+                override fun onStop(owner: LifecycleOwner) {
+                    unbindService()
+                }
+            },
+        )
+    }
+
+    private fun bindService() {
+        val intent = Intent().apply {
+            component = ComponentName(
+                connectionType.toPackageName(),
+                AUTHENTICATOR_BRIDGE_SERVICE_CLASS,
+            )
+        }
+
+        val isBound = try {
+            applicationContext.bindService(
+                intent,
+                bridgeServiceConnection,
+                Context.BIND_AUTO_CREATE,
+            )
+        } catch (e: SecurityException) {
+            unbindService()
+            false
+        }
+
+        if (!isBound) {
+            mutableSharedAccountsStateFlow.value = AccountSyncState.Error
+        }
+    }
+
+    private fun onAccountsSync(data: EncryptedSharedAccountData) {
+        // Received account sync update. Decrypt with local symmetric key and update StateFlow:
+        mutableSharedAccountsStateFlow.value = symmetricKeyStorageProvider.symmetricKey
+            ?.let { data.decrypt(it) }
+            ?.getOrNull()
+            ?.let { AccountSyncState.Success(it.accounts) }
+            ?: AccountSyncState.Error
+    }
+
+    private fun onServiceConnected(service: IBinder) {
+        bridgeService = IAuthenticatorBridgeService.Stub.asInterface(service)
+
+        // TODO: Add check for version mismatch between client and server SDKs: BITAU-72
+
+        // Ensure we are using the correct symmetric key:
+        val localKeyFingerprint =
+            symmetricKeyStorageProvider.symmetricKey?.toFingerprint()?.getOrNull()
+
+        // Query bridge service to see if we have a matching symmetric key:
+        val haveCorrectKey = bridgeService
+            .safeCall { checkSymmetricEncryptionKeyFingerprint(localKeyFingerprint) }
+            .fold(
+                onSuccess = { it },
+                onFailure = { false },
+            )
+            ?: false
+
+        if (!haveCorrectKey) {
+            // If we don't have the correct key, query for key:
+            symmetricKeyStorageProvider.symmetricKey =
+                bridgeService.safeCall { symmetricEncryptionKeyData }.getOrNull()
```

Comment:
I would think we'd want the failure state here to end up emitting `AccountSyncState.Error` or whatever it's called. Same thing above. I think this is a case where `null` is different from an error.